stack/pico_stack.c

```diff
@@ -857,6 +856,20 @@ uint32_t pico_timer_add_hashed(pico_time expire, void (*timer)(pico_time, void *
 
 int MOCKABLE pico_stack_init(void)
 {
+
+#ifdef CHECK_MEM
+	//New log for malloc information 
+	fopen("mem_test.log", "w");
```

Comment:
The idea was to make sure to clear the file before running a new set of tests but this was flawed since it was getting called with each new test and so only the last test's mallocs were getting logged in the end. I changed to outputting this in stdout, which also makes it clear which tests the mallocs occurred in and which ASAN errors are related to them.